### PR TITLE
RET 3842 hearing is linked flag

### PR DESCRIPTION
**Pull request by brendanmchugh**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RET-3842

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java [hunk 1]
```diff
@@ -74,6 +74,7 @@
 import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.CallbackRespHelper.getCallbackRespEntityErrors;
 import static uk.gov.hmcts.ethos.replacement.docmosis.helpers.CallbackRespHelper.getCallbackRespEntityNoErrors;
 
+@SuppressWarnings({"PMD.TooManyFields"})
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java:77]: Is this required? looks like you've not added any new fields to this class

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java [hunk 2]
```diff
@@ -56,14 +58,17 @@ public ResponseEntity<CCDCallbackResponse> createCaseLinkAboutToSubmit(
             return ResponseEntity.status(FORBIDDEN.value()).build();
         }
         log.info("Adding case links");
+        ccdRequest.getCaseDetails().getCaseData().setHearingIsLinkedFlag(Boolean.TRUE.toString());
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java:61]: I know the way we handle this in CCD is a little unconventional, but I'm thinking using "Yes" or "No" would be better here.. The other flags use this pattern and its more native to CCD fields

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java [hunk 3]
```diff
@@ -86,6 +91,11 @@ public ResponseEntity<CCDCallbackResponse> maintainCaseLinkAboutToSubmit(
         }
         log.info("Removing case links");
         caseLinksEmailService.sendMailWhenCaseLinkForHearing(ccdRequest, userToken, false);
+        CaseData caseData = ccdRequest.getCaseDetails().getCaseData();
+
+        if (caseData.getCaseLinks() == null || caseData.getCaseLinks().isEmpty()) {
+            caseData.setHearingIsLinkedFlag(Boolean.FALSE.toString());
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java:97]: Same here

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java [hunk 4]
```diff
@@ -162,6 +162,10 @@ public void setPublicCaseName(CaseData caseData) {
         }
     }
 
+    public void setHearingIsLinkedFlag(CaseData caseData) {
+        caseData.setHearingIsLinkedFlag(Boolean.FALSE.toString());
```
Comment (JackReevies) [inline src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java:166]: Same here

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksControllerTest.java [hunk 5]
```diff
@@ -120,4 +123,69 @@ void testInitTransferToEnglandWalesForbidden() throws Exception {
...
+        mockMvc.perform(post(CREATE_SUBMITTED_URL)
+                        .content(jsonMapper.toJson(ccdRequest))
+                        .header("Authorization", AUTH_TOKEN)
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.data", notNullValue()))
+                .andExpect(jsonPath("$.errors", nullValue()))
+                .andExpect(jsonPath("$.warnings", nullValue()))
+                .andExpect(jsonPath("$.data.hearingIsLinkedFlag").value(
+                "true"));
```
Comment (JackReevies) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksControllerTest.java:139]: Can this go on a single line?

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerServiceTest.java [hunk 6]
```diff
@@ -1069,6 +1069,13 @@ void testCaseDeepLink() {
         assertEquals("http://domain/cases/case-details/123#Documents", caseData.getCaseDeepLink());
     }
 
+    @Test
+    void testHearingIsLinkedFlag() {
+        CaseData caseData = new CaseData();
+        caseManagementForCaseWorkerService.setHearingIsLinkedFlag(caseData);
+        assertEquals(Boolean.FALSE.toString(), caseData.getHearingIsLinkedFlag());
+    }
+
```
Comment (JackReevies) [inline src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerServiceTest.java:1078]: Do we need a test here as well checking that it sets the flag to true/Yes ?

**Review by JackReevies — CHANGES_REQUESTED** [review]

**Review by lvo700 — APPROVED** [review]

File: src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerServiceTest.java (hunk 6 above)
Comment (brendanmchugh) [reply src/test/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerServiceTest.java:1078]: The caseManagementForCaseWorkerService only adds the flag and defaults it to false.
The createCaseLinkAboutToSubmit method (on CaseLinksController) changes it to yes.

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java (hunk 1 above)
Comment (brendanmchugh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java]: removed

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java (hunk 2 above)
Comment (brendanmchugh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java]: change to yes

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java (hunk 4 above)
Comment (brendanmchugh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/CaseManagementForCaseWorkerService.java]: updated

File: src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java (hunk 3 above)
Comment (brendanmchugh) [reply src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseLinksController.java]: updated

**Review by JackReevies — APPROVED** [review]

**Pull request merged**